trigger/kafkasub/trigger.go

```diff
@@ -0,0 +1,390 @@
+package kafkasub
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+
+	"fmt"
+	"net/url"
+
+	"time"
+
+	"crypto/tls"
+	"crypto/x509"
+
+	"io/ioutil"
+
+	"github.com/Shopify/sarama"
+	"github.com/TIBCOSoftware/flogo-lib/core/action"
+	"github.com/TIBCOSoftware/flogo-lib/core/trigger"
+	"github.com/TIBCOSoftware/flogo-lib/logger"
+)
+
+// log is the default package logger
+var flogoLogger = logger.GetLogger("trigger-tibco-kafkasub")
+
+type _kafkaParms struct {
+	brokers    []string
+	topic      string
+	offset     int64
+	group      string
+	partitions []int32
+}
+
+// MqttTrigger is simple MQTT trigger
+type KafkaSubTrigger struct {
+	metadata           *trigger.Metadata
+	runner             action.Runner
+	config             *trigger.Config
+	kafkaParms         _kafkaParms
+	shutdownChan       *chan struct{}
+	signals            *chan os.Signal
+	kafkaConfig        *sarama.Config
+	kafkaConsumer      *sarama.Consumer
+	partitionConsumers *map[int32]sarama.PartitionConsumer
+}
+
+//NewFactory create a new Trigger factory
+func NewFactory(md *trigger.Metadata) trigger.Factory {
+	return &KafkasubFactory{metadata: md}
+}
+
+// KafkasubFactory Kafka subscriber Trigger factory
+type KafkasubFactory struct {
+	metadata *trigger.Metadata
+}
+
+//New Creates a new trigger instance for a given id
+func (t *KafkasubFactory) New(config *trigger.Config) trigger.Trigger {
+	kafkaTrigger := &KafkaSubTrigger{metadata: t.metadata, config: config}
+	kafkaTrigger.kafkaConfig = sarama.NewConfig()
+	kafkaTrigger.kafkaConfig.Consumer.Return.Errors = true
+	error := initKafkaParms(kafkaTrigger)
+	if error != nil {
+		flogoLogger.Error("Failed to initialize parms: ", error)
+		return nil
```

Comment:
This will fail silently, I think it would it be better to just panic() for now to fail fast instead of returning nil.